### PR TITLE
Added some returning but not documented AddressType.

### PR DESCRIPTION
**Pull request by onlyonce**
I've added some returning but not documented AddressType from https://developers.google.com/places/documentation/supported_types, and optimized the lookup method.
I've made some basic coding style corrections (add serialVersionUID; logger variables be static, etc).

### CODE REVIEW

File: src/main/java/com/google/maps/model/AddressType.java [hunk 1]
```diff
@@ -183,16 +188,44 @@
...
+  PLACE_OF_WORSHIP("place_of_worship"),
+    
+  /**
    * Indicates an unknown address type returned by the server. The Java Client for Google Maps
    * Services should be updated to support the new value.
    */
   UNKNOWN("unknown");
 
-  private static Logger log = Logger.getLogger(AddressType.class.getName());
+  private static final Logger log = Logger.getLogger(AddressType.class.getName());
```
Comment (broady) [inline src/main/java/com/google/maps/model/AddressType.java]: `LOG` for consistency?

File: src/main/java/com/google/maps/model/AddressType.java [hunk 2]
```diff
@@ -208,74 +241,20 @@ public String toUrlValue() {
...
+   * @return specified AddressType enum value or if it's not found then {@value AddressType#UNKNOWN}.
+   */
+  public static AddressType lookup(final String addressType) {
+      AddressType result = VALUE_MAP.get(addressType.toLowerCase(Locale.ENGLISH));
+      if (null != result) {
+          return result;
+      }
 
-  public static AddressType lookup(String addressType) {
-    if (addressType.equalsIgnoreCase(STREET_ADDRESS.toString())) {
```
Comment (broady) [inline src/main/java/com/google/maps/model/AddressType.java]: I think this class was missed in change e2aace0c337. Should use the same approach, I think.

**Comment by onlyonce** [comment]
OK! I've missed the log in the AddressType.
I've removed the lookup() method and the logger from the AddressType.
I've commited to my master. I'm very newbie in the GitHub, please help, what I need to do?

File: src/main/java/com/google/maps/model/AddressType.java [hunk 3]
```diff
@@ -15,15 +15,20 @@
...
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import com.google.maps.internal.StringJoin.UrlValue;
+
 /**
  * The Adress types. Please see
  * <a href="https://developers.google.com/maps/documentation/geocoding/#Types">Address
  * Types</a> for more detail.
+ * Some address contains additional place category. Please see
```
Comment (markmcd) [inline src/main/java/com/google/maps/model/AddressType.java]: "Some addresses contain additional place categories."

While you're in here, would you mind fixing the type on line 27 too? "The Adress types" should read "The *address* types"

Comment (onlyonce) [inline src/main/java/com/google/maps/model/AddressType.java]: Done!

**Comment by broady** [comment]
LGTM - @markmcd want to take a final look?

File: src/main/java/com/google/maps/model/AddressType.java [hunk 4]
```diff
@@ -15,15 +15,20 @@
...
 import java.util.logging.Logger;
 
+import com.google.maps.internal.StringJoin.UrlValue;
+
 /**
- * The Adress types. Please see
+ * The Address types. Please see
  * <a href="https://developers.google.com/maps/documentation/geocoding/#Types">Address
  * Types</a> for more detail.
+ * Some address contains additional place category. Please see
```
Comment (markmcd) [inline src/main/java/com/google/maps/model/AddressType.java]: Just a small one. As before - this first sentence should read: "Some addresses contain additional place categories."

Comment (onlyonce) [inline src/main/java/com/google/maps/model/AddressType.java]: done!

**Comment by markmcd** [comment]
LGTM!

**Pull request merged**